src/endpoints/auth/check-auth.route.ts

```diff
@@ -0,0 +1,48 @@
+import { jwtConfig } from '../../shared/configs';
+import { USER_STATE_COOKIE } from '../../shared/consts';
+import { UserEntity } from '../../shared/database';
+import { UserStateEnum } from '../../shared/enums';
+import { UnauthorizedException } from '../../shared/errors';
+import { JWTService } from '../../shared/services';
+import { RouteCustomOptions } from '../../shared/types';
+import { checkAuthSchema, ICheckAuthBodySchema as ICheckAuthBodyInput } from './schemas';
+
+const { secret, accessDeathDate, refreshDeathDate } = jwtConfig;
+
+export const checkAuthRoute: RouteCustomOptions<{ Body: ICheckAuthBodyInput }> = {
+  url: '/check-auth',
+  method: 'POST',
+  schema: checkAuthSchema,
+  handler: async (req, reply) => {
+    const { refreshToken: inputRefreshToken } = req.body;
+
+    const tokenData = await JWTService.checkRefreshToken(secret, inputRefreshToken);
+
+    if (!tokenData) {
+      throw UnauthorizedException('Invalid refresh token.');
+    }
+
+    const user = await UserEntity.findOneBy({ sessionKey: tokenData.sessionKey });
+
+    if (!user) {
+      throw UnauthorizedException('Invalid refresh token.');
+    }
+
+    reply.setCookie(USER_STATE_COOKIE, user.state || UserStateEnum.NOT_VERIFIED);
+
+    const { accessToken, refreshToken, accessTokenExpiresIn, refreshTokenExpiresIn } =
+      await JWTService.generateTokens({
+        sessionKey: tokenData.sessionKey,
+        jwtSecret: secret,
+        accessDeathDate,
+        refreshDeathDate,
+      });
+
+    return {
```

Comment:
Are you sure your `setCookie` works when you simply `return` your response and not `reply.send()` it? Have you tried it?